torch/csrc/jit/tensorexpr/analysis.h

```diff
@@ -248,6 +248,194 @@ class ModifiesVarChecker : public IRVisitor {
   bool found_{false};
 };
 
+// This indicates the path from root to a stmt A in AST. Specifically, it
+// consists of a vector of integers: each integer represents the number of stmts
+// before A's ancestor or A in the block. For example, "1:1:0" points to stmt
+// "d[i] += e[i, j];" in the following code.
+//
+// c = a/b;
+// for i
+//  d[i] = 0;
+//  for j
+//    d[i] += e[i, j];
+class StmtIndex {
+ public:
+  void append(int32_t id) {
+    stmt_index_.emplace_back(id);
+  }
+  void pop() {
+    stmt_index_.pop_back();
+  }
+
+  const std::vector<int32_t>& getStmtIndex() {
+    return stmt_index_;
+  }
+
+  bool operator==(StmtIndex& index) {
+    auto compare = index.getStmtIndex();
+    if (stmt_index_.size() != compare.size()) {
+      return false;
+    }
+    int size = compare.size();
+    for (int i = 0; i < size; i++) {
+      if (stmt_index_.at(i) != compare.at(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool operator<(StmtIndex& index) {
+    auto compare = index.getStmtIndex();
+    int size = stmt_index_.size() < compare.size() ? stmt_index_.size()
+                                                   : compare.size();
+    for (int i = 0; i < size; i++) {
+      if (stmt_index_.at(i) > compare.at(i)) {
+        return false;
+      }
+    }
+    return !(*this == index);
+  }
```

Comment:
Since we're no longer looking into inner statements in mem planning, we're no longer using these comparisons, right? If so, I'd prefer getting rid of these methods and consider removing this class altogether.  getStmtIndexString method can be made a static free-standing function.

We need to optimize our code for (future) readers, not for (present) writers: i.e. even if the code we're adding now might look better with these classes, we need to take into account how difficult it would be to grasp it in future for people unfamiliar with it. That's why I generally prefer not to introduce wrappers over standard classes - we can assume that everyone knows API of std::vector, but we can't assume people know API of a custom class we've just added (as an example: can you tell what methods are there in `AccessInfo` without looking at the code?).